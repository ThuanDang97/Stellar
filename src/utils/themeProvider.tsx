import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { theme } from '@themes/index'

export const renderWithTheme = (component: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}
