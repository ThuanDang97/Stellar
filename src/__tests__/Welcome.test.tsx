// Pages
import Welcome from '@pages/welcome'

// Utils
import { renderWithTheme } from '@utils/themeProvider'

describe('Welcome Component', () => {
  test('Component Welcome matches DOM Snapshot', () => {
    const { container } = renderWithTheme(<Welcome />)

    expect(container).toMatchSnapshot()
  })
})
