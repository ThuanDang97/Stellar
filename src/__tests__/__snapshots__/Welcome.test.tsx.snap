// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Component Component Welcome matches DOM Snapshot 1`] = `
<div>
  <div
    class="ViewStyled-sc-rqe5sa-0 AnimatedPageStyled__PageFadeOut-sc-4qvv2f-1 kuBhDh jxegJD"
    width="100%"
  >
    <div
      class="ViewStyled-sc-rqe5sa-0 LayoutStyled__SecondaryLayout-sc-ivdxqy-1 fTsvHV jlGBer"
    >
      <div
        class="ViewStyled-sc-rqe5sa-0 ViewStyled__WrapperContent-sc-rqe5sa-6 kruvYh gVWyht"
        display="flex"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 40px; height: 40px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <img
            data-nimg="fixed"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <h1
          class="TextStyled-sc-9xia41-0 TitleStyled-sc-t58pjn-0 khnHTb jSOOxW"
          color="#FFF"
          font-size="36"
          font-weight="700"
          width="min(90%, 19.6875rem)"
        >
          Welcome to Stellar,
        </h1>
        <h1
          class="TextStyled-sc-9xia41-0 TitleStyled-sc-t58pjn-0 cwpvxF jSOOxW"
          color="#FFF"
          font-size="36"
          font-weight="700"
          overflow="hidden"
          width="min(90%, 19.6875rem)"
        />
      </div>
    </div>
  </div>
</div>
`;
